client/components/editor/teaching-elements/Assessment/ImageMultipleChoice.vue

```diff
@@ -0,0 +1,326 @@
+      } else {
+        correct.splice(position, 1);
+      }
+
+      this.update({ correct });
+    },
+    updateAnswer(evt) {
+      let answers = cloneDeep(this.answers);
+      const { target: input } = evt;
+      const { dataset, files = [] } = input;
+      const imageFile = files[0];
+
+      return new Promise((resolve, reject) => {
+        this.resetErrorMessages(dataset.index);
+        if (imageFile.size > 100000) {
+          return reject(new ImageValidationError(
+            'The image is too large. Maximum allowed image size: 100 kB'
+          ));
+        }
+        return resolve(imageFile);
+      })
+        .then(() => blobToDataURL(imageFile))
+        .then(imageDataUrl => {
+          answers[dataset.index] = imageDataUrl;
+          return getImageDimensions(imageDataUrl);
+        })
+        .then(({ width, height }) => {
+          if (width > maxImageDimension || height > maxImageDimension) {
+            return Promise.reject(new ImageValidationError(
+              `Incorrect image width and/or height. Maximum allowed image dimensions:
+              ${maxImageDimension}x${maxImageDimension}`
+            ));
+          }
+        })
+        .catch(error => {
+          answers[dataset.index] = '';
+          if (error.name === 'ImageValidationError') {
+            this.addErrorMessage(dataset.index, error.message);
+          } else {
+            throw error;
+          }
+        })
+        .then(() => this.update({ answers }));
+    },
+    addAnswer() {
+      let answers = cloneDeep(this.answers);
+      answers.push('');
+      this.update({ answers });
+    },
+    removeAnswer(answerIndex) {
+      let answers = cloneDeep(this.answers);
+      let correct = cloneDeep(this.correct);
+      let feedback = cloneDeep(this.feedback);
+
+      answers.splice(answerIndex, 1);
+      const index = correct.indexOf(answerIndex);
+      if (index !== -1) correct.splice(index, 1);
+
+      correct.forEach((it, i) => {
+        if (it >= answerIndex) correct[i] = it - 1;
+      });
+
+      if (feedback) {
+        range(answerIndex, answers.length).forEach(it => {
+          feedback[it] = feedback[it + 1];
+        });
+        delete feedback[answers.length];
+      }
+
+      this.fieldErrors.splice(answerIndex, 1);
+
+      this.update({ answers, correct, feedback });
+    },
+    hasQuestionText() {
+      let questionContent = this.assessment.question[0].data.content || '';
+      return questionContent !== '';
+    },
+    hasCorrectAnswers() {
+      return this.correct.length > 0;
+    },
+    validate() {
+      let error = {};
+      if (!this.hasQuestionText()) {
+        error = formAlert('The question field may not be empty.');
+      } else if (this.answers.length < 2) {
+        error = formAlert('Please make at least two answers available.');
+      } else if (!this.hasCorrectAnswers()) {
+        error = formAlert('There needs to be at least one correct answer.');
+      }
```

Comment:
Take this as _rule of thumb_. You never repeat same function call in different execution paths, instead you do it once with arguments defined by those paths.

Call `formAlert` once, and use these blocks to define its `error` argument.